sdk/Pulumi/Deployment/Deployment.cs

```diff
@@ -183,8 +183,10 @@ internal async Task<CallbacksHost> GetCallbacksAsync(CancellationToken cancellat
                 return _callbacks;
             }
 
-            // Atomically allocate a callbacks instance to use
-            var callbacks = new CallbacksHost();
+            // Atomically allocate a callbacks instance to use, capture the current async context and pass it
+            // to the callbacks host. Importantly this captures the current Deployment.Instance, so basic
+            // things like engine logging work in transforms.
+            var callbacks = new CallbacksHost(System.Threading.ExecutionContext.Capture());
```

Comment:
Nit: Aren't we already importing `System.Threading`? Is there some other reason it needs to be fully qualified?

```suggestion
            var callbacks = new CallbacksHost(ExecutionContext.Capture());
```